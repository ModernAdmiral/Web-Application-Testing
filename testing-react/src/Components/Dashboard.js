import React, { useState } from "react";
import Display from "./Display";

const Dashboard = () => {
  const [ball, setBall] = useState(0);
  const [strike, setStrike] = useState(0);
  return (
    <>
      <Display balls={ball} strikes={strike} />
      <h1>Dashboard</h1>
      <button
        onClick={() => {
          setBall(ball + 1);
        }}
      >
        ball!
      </button>
      <button
        onClick={() => {
          setStrike(strike + 1);
        }}
      >
        strike!
      </button>
    </>
  );
};

export default Dashboard;
