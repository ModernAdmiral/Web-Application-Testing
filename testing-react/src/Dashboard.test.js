import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Dashboard from "./Components/Dashboard";

afterEach(cleanup);
it("renders", () => {
  const { asFragment } = render(<Dashboard balls={4} strikes={2} />);
  expect(asFragment()).toMatchSnapshot();
});
