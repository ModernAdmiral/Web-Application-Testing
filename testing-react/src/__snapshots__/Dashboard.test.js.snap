// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h1>
    Display
  </h1>
  <div>
    <h2>
      Balls: 0
    </h2>
    <h2>
      Strikes: 0
    </h2>
  </div>
  <h1>
    Dashboard
  </h1>
  <button>
    ball!
  </button>
  <button>
    strike!
  </button>
</DocumentFragment>
`;
